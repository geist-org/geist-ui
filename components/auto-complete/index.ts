import AutoComplete from './auto-complete'
import AutoCompleteItem from './auto-complete-item'
import AutoCompleteSearching from './auto-complete-searching'
import AutoCompleteEmpty from './auto-complete-empty'

export type AutoCompleteComponentType = typeof AutoComplete & {
  Item: typeof AutoCompleteItem
  Option: typeof AutoCompleteItem
  Searching: typeof AutoCompleteSearching
  Empty: typeof AutoCompleteEmpty
}
;(AutoComplete as AutoCompleteComponentType).Item = AutoCompleteItem
;(AutoComplete as AutoCompleteComponentType).Option = AutoCompleteItem
;(AutoComplete as AutoCompleteComponentType).Searching = AutoCompleteSearching
;(AutoComplete as AutoCompleteComponentType).Empty = AutoCompleteEmpty

export type {
  AutoCompleteOption,
  AutoCompleteOptions,
  AutoCompleteProps,
  AutoCompleteTypes,
} from './auto-complete'
export default AutoComplete as AutoCompleteComponentType
