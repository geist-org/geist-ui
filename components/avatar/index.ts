import Avatar from './avatar'
import AvatarGroup from './avatar-group'

export type AvatarComponentType = typeof Avatar & {
  Group: typeof AvatarGroup
}
;(Avatar as AvatarComponentType).Group = AvatarGroup

export type { AvatarProps } from './avatar'
export type { AvatarGroupProps } from './avatar-group'

export default Avatar as AvatarComponentType
