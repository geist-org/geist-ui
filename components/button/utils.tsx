import React, { ReactNode } from 'react'
import ButtonIcon from './button-icon'
import { ButtonProps } from './button'
import { ButtonGroupConfig } from '../button-group/button-group-context'
import useClasses from '../use-classes'

export const getButtonChildrenWithIcon = (
  auto: boolean,
  children: ReactNode,
  icons: {
    icon?: React.ReactNode
    iconRight?: React.ReactNode
  },
) => {
  const { icon, iconRight } = icons
  const hasIcon = icon || iconRight
  const isRight = Boolean(iconRight)
  const paddingForAutoMode = auto
    ? `calc(var(--geist-ui-button-height) / 2 + var(--geist-ui-button-icon-padding) * .5)`
    : 0
  const classes = useClasses('text', isRight ? 'right' : 'left')

  if (!hasIcon) return <div className="text">{children}</div>
  if (React.Children.count(children) === 0) {
    return (
      <ButtonIcon isRight={isRight} isSingle>
        {hasIcon}
      </ButtonIcon>
    )
  }
  return (
    <>
      <ButtonIcon isRight={isRight}>{hasIcon}</ButtonIcon>
      <div className={classes}>
        {children}
        <style jsx>{`
          .left {
            padding-left: ${paddingForAutoMode};
          }
          .right {
            padding-right: ${paddingForAutoMode};
          }
        `}</style>
      </div>
    </>
  )
}

export const filterPropsWithGroup = <T extends React.PropsWithChildren<ButtonProps>>(
  props: T,
  config: ButtonGroupConfig,
): T => {
  if (!config.isButtonGroup) return props
  return {
    ...props,
    auto: true,
    shadow: false,
    ghost: config.ghost || props.ghost,
    type: config.type || props.type,
    disabled: config.disabled || props.disabled,
  }
}
