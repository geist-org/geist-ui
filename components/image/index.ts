import Image from './image'
import ImageBrowser from './image-browser'

export type ImageComponentType = typeof Image & {
  Browser: typeof ImageBrowser
}
;(Image as ImageComponentType).Browser = ImageBrowser

export type { ImageProps } from './image'
export type { ImageBrowserProps, ImageAnchorProps } from './image-browser'
export default Image as ImageComponentType
