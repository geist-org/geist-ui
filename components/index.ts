// @ts-ignore
/// <reference types="styled-jsx" />

export { default as AutoComplete } from './auto-complete'
export type { AutoCompleteProps } from './auto-complete'

export { default as Avatar } from './avatar'
export type { AvatarProps, AvatarGroupProps } from './avatar'

export { default as Badge } from './badge'
export type { BadgeProps, BadgeAnchorProps } from './badge'

export { default as Breadcrumbs } from './breadcrumbs'
export type {
  BreadcrumbsProps,
  BreadcrumbsItemProps,
  BreadcrumbsSeparatorProps,
} from './breadcrumbs'

export { default as Button } from './button'
export type { ButtonProps } from './button'

export { default as ButtonDropdown } from './button-dropdown'
export type { ButtonDropdownProps, ButtonDropdownItemProps } from './button-dropdown'

export { default as ButtonGroup } from './button-group'
export type { ButtonGroupProps } from './button-group'

export { default as Capacity } from './capacity'
export type { CapacityProps } from './capacity'

export { default as Card } from './card'
export type { CardProps, CardContentProps, CardFooterProps } from './card'

export { default as Checkbox } from './checkbox'
export type { CheckboxProps, CheckboxGroupProps } from './checkbox'

export { default as Code } from './code'
export type { CodeProps } from './code'

export { default as Collapse } from './collapse'
export type { CollapseProps, CollapseGroupProps } from './collapse'

export { default as Description } from './description'
export type { DescriptionProps } from './description'

export { default as Display } from './display'
export type { DisplayProps } from './display'

export { default as Divider } from './divider'
export type { DividerProps } from './divider'

export { default as Dot } from './dot'
export type { DotProps } from './dot'

export { default as Drawer } from './drawer'
export type { DrawerProps } from './drawer'

export { default as Fieldset } from './fieldset'
export type {
  FieldsetProps,
  FieldsetTitleProps,
  FieldsetSubtitleProps,
  FieldsetGroupProps,
  FieldsetFooterProps,
  FieldsetContentProps,
} from './fieldset'

export { default as GeistProvider } from './geist-provider'
export type { GeistProviderProps } from './geist-provider'

export { default as Grid } from './grid'
export type { GridProps, GridContainerProps } from './grid'

export { default as Image } from './image'
export type { ImageProps, ImageBrowserProps } from './image'

export { default as Input } from './input'
export type { InputProps, InputPasswordProps } from './input'

export { default as Keyboard } from './keyboard'
export type { KeyboardProps } from './keyboard'

export { default as Link } from './link'
export type { LinkProps } from './link'

export { default as Loading } from './loading'
export type { LoadingProps } from './loading'

export { default as Modal } from './modal'
export type {
  ModalProps,
  ModalTitleProps,
  ModalSubtitleProps,
  ModalContentProps,
  ModalActionProps,
} from './modal'

export { default as Note } from './note'
export type { NoteProps } from './note'

export { default as Page } from './page'
export type {
  PageProps,
  PageHeaderProps,
  PageContentProps,
  PageFooterProps,
} from './page'

export { default as Pagination } from './pagination'
export type {
  PaginationProps,
  PaginationNextProps,
  PaginationPreviousProps,
} from './pagination'

export { default as Popover } from './popover'
export type { PopoverProps, PopoverItemProps } from './popover'

export { default as Progress } from './progress'
export type { ProgressProps } from './progress'

export { default as Radio } from './radio'
export type { RadioProps, RadioGroupProps, RadioDescriptionProps } from './radio'

export { default as Rating } from './rating'
export type { RatingProps } from './rating'

export { default as Select } from './select'
export type { SelectProps, SelectOptionProps } from './select'

export { default as Slider } from './slider'
export type { SliderProps } from './slider'

export { default as Snippet } from './snippet'
export type { SnippetProps } from './snippet'

export { default as Spacer } from './spacer'
export type { SpacerProps } from './spacer'

export { default as Spinner } from './spinner'
export type { SpinnerProps } from './spinner'

export { default as Table } from './table'
export type { TableProps, TableColumnProps } from './table'

export { default as Tabs } from './tabs'
export type { TabsProps } from './tabs'

export { default as Tag } from './tag'
export type { TagProps } from './tag'

export { default as Text } from './text'
export type { TextProps } from './text'

export { default as Textarea } from './textarea'
export type { TextareaProps } from './textarea'

export { default as Themes } from './themes'
export type { GeistUIThemes, GeistUserTheme } from './themes'

export { default as Toggle } from './toggle'
export type { ToggleProps } from './toggle'

export { default as Tooltip } from './tooltip'
export type { TooltipProps } from './tooltip'

export { default as Tree } from './tree'
export type { TreeProps } from './tree'

export { default as useAllThemes } from './use-all-themes'
export type { AllThemesConfig } from './use-all-themes'

export { default as useToasts } from './use-toasts'
export type { Toast, ToastInput, ToastAction, ToastLayout } from './use-toasts'

export { default as User } from './user'
export type { UserProps } from './user'

export { default as useBodyScroll } from './use-body-scroll'
export type { BodyScrollOptions } from './use-body-scroll'

export { default as useClipboard } from './use-clipboard'
export type { UseClipboardOptions } from './use-clipboard'

export { default as useMediaQuery } from './use-media-query'
export type { ResponsiveOptions, ResponsiveBreakpoint } from './use-media-query'

export { default as useKeyboard, KeyMod, KeyCode } from './use-keyboard'
export type { KeyboardOptions, UseKeyboardHandler } from './use-keyboard'

export { default as useInput } from './use-input'
export { default as useModal } from './use-modal'
export { default as useTabs } from './use-tabs'
export { default as useClickAway } from './use-click-away'
export { default as useCurrentState } from './use-current-state'
export { default as CssBaseline } from './css-baseline'
export { default as useTheme } from './use-theme'
export { default as useClasses } from './use-classes'
export { default as useScale } from './use-scale'
export { withScale, ScalePropKeys, ScaleContext } from './use-scale'
export type {
  ScaleProps,
  ScaleConfig,
  GetScalePropsFunction,
  GetAllScalePropsFunction,
} from './use-scale'

export { useRect } from './utils/layouts'
export type { ReactiveDomReact } from './utils/layouts'
export { default as Highlight } from './shared/highlight'
export type { HighlightProps } from './shared/highlight'
