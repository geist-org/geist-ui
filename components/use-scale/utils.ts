import {
  GetAllScalePropsFunction,
  GetScalePropsFunction,
  ScaleProps,
  ScalePropKeys,
} from './scale-context'

export const generateGetScaleProps = <P>(
  props: P & ScaleProps,
): GetScalePropsFunction => {
  const getScaleProps: GetScalePropsFunction = keyOrKeys => {
    if (!Array.isArray(keyOrKeys)) return props[keyOrKeys as keyof ScaleProps]
    let value = undefined
    for (const key of keyOrKeys) {
      const currentValue = props[key]
      if (typeof currentValue !== 'undefined') {
        value = currentValue
      }
    }
    return value
  }
  return getScaleProps
}

export const generateGetAllScaleProps = <P>(
  props: P & ScaleProps,
): GetAllScalePropsFunction => {
  const getAllScaleProps: GetAllScalePropsFunction = () => {
    let scaleProps: ScaleProps = {}
    for (const key of ScalePropKeys) {
      const value = props[key as keyof ScaleProps]
      if (typeof value !== 'undefined') {
        scaleProps[key as keyof ScaleProps] = value as any
      }
    }
    return scaleProps
  }
  return getAllScaleProps
}
