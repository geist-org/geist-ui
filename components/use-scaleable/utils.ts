import { ScaleableProps } from './scaleable-context'

export type ScaleablePropsAndInvalid = keyof ScaleableProps | 'size'

export const ScaleablePropKeys: Array<ScaleablePropsAndInvalid> = [
  'paddingLeft',
  'pl',
  'paddingRight',
  'pr',
  'paddingTop',
  'pt',
  'paddingBottom',
  'pb',
  'marginTop',
  'mt',
  'marginRight',
  'mr',
  'marginBottom',
  'mb',
  'marginLeft',
  'ml',
  'px',
  'py',
  'mx',
  'my',
  'width',
  'height',
  'font',
  'unit',
  'scale',
  'size',
]

export const filterScaleableProps = <T extends Record<any, any>>(props: T) => {
  const keys = Object.keys(props).filter(key => key !== '')
  const nextProps: any = {}
  for (const key of keys) {
    if (!(ScaleablePropKeys as string[]).includes(key)) {
      nextProps[key] = props[key]
    }
  }
  return nextProps as Omit<T, ScaleablePropsAndInvalid>
}
