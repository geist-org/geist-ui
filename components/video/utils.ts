export const formatTime = (seconds: number) => {
  const hh = Math.floor(seconds / 3600)
  const mm = Math.floor(seconds / 60) % 60
  const ss = Math.floor(seconds) % 60
  const pad = (num: number) => (num < 10 ? '0' + num : num)
  return `${hh ? pad(hh) + ':' : ''}${pad(mm)}:${pad(ss)}`
}

export const isFullscreenEnabled = () => {
  const elem = document as any
  return (
    elem.fullscreenEnabled ||
    elem.webkitFullscreenEnabled ||
    elem.mozFullScreenEnabled ||
    elem.msFullscreenEnabled
  )
}

export const openFullscreen = (element: HTMLElement) => {
  const elem = element as any
  if (elem.requestFullscreen) {
    elem.requestFullscreen()
  } else if (elem.mozRequestFullScreen) {
    elem.mozRequestFullScreen()
  } else if (elem.webkitRequestFullscreen) {
    elem.webkitRequestFullscreen()
  } else if (elem.msRequestFullscreen) {
    elem.msRequestFullscreen()
  }
}

export const closeFullscreen = () => {
  const elem = document as any
  if (elem.exitFullscreen) {
    elem.exitFullscreen()
  } else if (elem.mozCancelFullScreen) {
    elem.mozCancelFullScreen()
  } else if (elem.webkitExitFullscreen) {
    elem.webkitExitFullscreen()
  } else if (elem.msExitFullscreen) {
    elem.msExitFullscreen()
  }
}

export const onFullscreenChange = (func: (state: boolean) => void) => {
  const doc = document as any
  doc.addEventListener('fullscreenchange', () => func(!!doc.fullscreen), true)
  doc.addEventListener('mozfullscreenchange', () => func(!!doc.mozFullScreen), true)
  doc.addEventListener('webkitfullscreenchange', () => func(!!doc.webkitIsFullScreen), true)
  doc.addEventListener('msfullscreenchange', () => func(!!doc.msFullscreenElement), true)
}

export const isSafari = () =>
  navigator.userAgent.includes('Safari') && !navigator.userAgent.includes('Chrome')
