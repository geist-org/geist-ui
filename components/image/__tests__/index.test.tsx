import React from 'react'
import { mount } from 'enzyme'
import { Image } from 'components'
import { updateWrapper } from 'tests/utils'

const url =
  'data:image/png;base64, iVBORw0KGgoAAAANSUhEUgAAAAUA' +
  'AAAFCAYAAACNbyblAAAAHElEQVQI12P4//8/w38GIAXDIBKE0DHxgljNBAAO' +
  '9TXL0Y4OHwAAAABJRU5ErkJggg=='

const svg = `<svg xmlns="http://www.w3.org/2000/svg" width="15px" height="15px" viewBox="0 0 15 15" fill="none">
      <path d="M12.5 8V7.83333C12.5 7.09695 11.903 6.5 11.1667 6.5H10C9.17157 6.5 8.5 7.17157 8.5 8C8.5 8.82843 9.17157 9.5 10 9.5H11C11.8284 9.5 12.5 10.1716 12.5 11C12.5 11.8284 11.8284 12.5 11 12.5H10C9.17157 12.5 8.5 11.8284 8.5 11M8 6.5H3M5.5 6.5V13M0.5 0.5H14.5V14.5H0.5V0.5Z" stroke="black"/></svg>`

describe('Image', () => {
  it('should render correctly', async () => {
    let wrapper = mount(<Image src={url} />)
    expect(wrapper).toMatchSnapshot()
    expect(() => wrapper.unmount()).not.toThrow()

    wrapper = mount(<Image src={url} width={20} height={20} />)
    wrapper.find('img').at(0).simulate('load')
    expect(wrapper).toMatchSnapshot()
    expect(() => wrapper.unmount()).not.toThrow()

    wrapper = mount(<Image src={url} width={20} height={20} disableSkeleton />)
    wrapper.find('img').at(0).simulate('load')
    expect(wrapper).toMatchSnapshot()
    expect(() => wrapper.unmount()).not.toThrow()
  })

  it('should render correctly with svg string', () => {
    let wrapper = mount(<Image src={svg} />)
    expect(wrapper).toMatchSnapshot()
    expect(() => wrapper.unmount()).not.toThrow()

    wrapper = mount(<Image src={svg} width={20} height={20} />)
    wrapper.find('img').at(0).simulate('load')
    expect(wrapper).toMatchSnapshot()
    expect(() => wrapper.unmount()).not.toThrow()

    wrapper = mount(<Image src={svg} width={20} height={20} disableSkeleton />)
    wrapper.find('img').at(0).simulate('load')
    expect(wrapper).toMatchSnapshot()
    expect(() => wrapper.unmount()).not.toThrow()
  })

  it('should work correctly with skeleton', async () => {
    let wrapper = mount(<Image src={url} width={20} height={20} />)
    expect(wrapper.find('.skeleton').length).not.toBe(0)

    wrapper = mount(<Image src={url} width={20} height={20} />)
    wrapper.find('img').at(0).simulate('load')
    await updateWrapper(wrapper)
    expect(wrapper.find('.skeleton').length).not.toBe(0)

    wrapper = mount(<Image src={url} width={20} />)
    expect(wrapper.find('.skeleton').length).toBe(0)

    mount(<Image src={url} width={20} height={20} disableSkeleton />)
    expect(wrapper.find('.skeleton').length).toBe(0)
  })

  it('should remove skeleton when timeout', async () => {
    const animation = mount(<Image src={url} width={20} height={20} maxDelay={100} />)
    const NoAnimation = mount(
      <Image src={url} width={20} height={20} maxDelay={100} disableSkeleton />,
    )
    expect(animation.find('.skeleton').length).not.toBe(0)
    await updateWrapper(animation, 300)
    await updateWrapper(NoAnimation, 300)
    expect(animation.find('.skeleton').length).toBe(0)
    expect(NoAnimation.find('.skeleton').length).toBe(0)
  })

  it('should remove skeleton when image completed', async () => {
    Object.defineProperty((global as any).Image.prototype, 'complete', {
      get() {
        return true
      },
    })
    const wrapper = mount(<Image src={url} width={20} height={20} />)
    const img = wrapper.find('img').at(0)
    img.simulate('load')
    await updateWrapper(wrapper)
    expect((img.getDOMNode() as HTMLImageElement).complete).toEqual(true)
    expect(wrapper.find('.skeleton').length).toBe(0)
  })
})
