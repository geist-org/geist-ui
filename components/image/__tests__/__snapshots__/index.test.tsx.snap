// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image should render correctly 1`] = `ReactWrapper {}`;

exports[`Image should render correctly 2`] = `ReactWrapper {}`;

exports[`Image should render correctly 3`] = `ReactWrapper {}`;

exports[`Image should render correctly with svg string 1`] = `ReactWrapper {}`;

exports[`Image should render correctly with svg string 2`] = `ReactWrapper {}`;

exports[`Image should render correctly with svg string 3`] = `ReactWrapper {}`;
